src/main/java/fi/aalto/cs/apluscourses/presentation/module/ModuleListElementViewModel.java

```diff
@@ -34,35 +36,35 @@ public String getStatus() {
     Module model = getModel();
     switch (model.stateMonitor.get()) {
       case Component.UNRESOLVED:
-        return "Unknown";
+        return getText("presentation.moduleStatuses.unknown");
       case Component.NOT_INSTALLED:
-        return "Double-click to download";
+        return getText("presentation.moduleStatuses.notInstalled");
       case Component.FETCHING:
-        return "Downloading...";
+        return getText("presentation.moduleStatuses.fetching");
       case Component.FETCHED:
-        return "Double-click to install";
+        return getText("presentation.moduleStatuses.fetched");
       case Component.LOADING:
-        return "Installing...";
+        return getText("presentation.moduleStatuses.loading");
       case Component.LOADED:
         break;
       case Component.UNINSTALLING:
-        return "Removing...";
+        return getText("presentation.moduleStatuses.uninstalling");
       case Component.UNINSTALLED:
-        return "Removed";
+        return getText("presentation.moduleStatuses.uninstalled");
       case Component.ACTION_ABORTED:
-        return "Cancelling...";
+        return getText("presentation.moduleStatuses.actionAborted");
       default:
-        return "Error";
+        return getText("presentation.moduleStatuses.error");
     }
     switch (model.dependencyStateMonitor.get()) {
       case Component.DEP_INITIAL:
-        return "Installed, dependencies unknown";
+        return getText("presentation.dependencyStatus.depInitial");
       case Component.DEP_WAITING:
-        return "Waiting for dependencies...";
+        return getText("presentation.dependencyStatus.depWaiting");
       case Component.DEP_LOADED:
-        return "Installed";
+        return getText("presentation.dependencyStatus.depLoaded");
       default:
-        return "Error in dependencies";
+        return getText("presentation.dependencyStatus.depError");
```

Comment:
Nit: I'd perhaps prefer `getText()` taken out of the `switch` block. If we wanted to keep `return`s inside `case`s, we could introduce a private function that solves the key string using `switch ... case` and the public method woukd call that and pass its return value to `getText`.

On the other hand, it's also a good thing that key strings doesn't appear in our code elsewhere but strictly lexically surrounded by `getText(...)`. I don't even know, if compiler can actually inline those, in which case this would be a very well-justified approach.